src/wrapper.jl

```diff
@@ -482,24 +550,26 @@ end
 
 function LGBM_BoosterPredictForMat(bst::Booster, data::Matrix{T},
                                                               predict_type::Integer,
+                                                              start_iteration::Integer,
                                                               num_iteration::Integer,
                                                               is_row_major::Bool = false) where T<:Union{Float32,Float64}
     num_class = LGBM_BoosterGetNumClasses(bst)
-    lgbm_data_type = jltype_to_lgbmid(T)
+    data_type = jltype_to_lgbmid(T)
```

Comment:
There wasn't a need to change this one, right? You don't need to change it back, it just looks to me like it wasn't necessary

Reply:
it was... C-API changed [see_changes](https://lightgbm.readthedocs.io/en/latest/C-API.html#c.LGBM_DatasetCreateFromMats) and this parameter got renamed which eventually caused segfaulting

Reply:
Are you sure it wasn't in connection with other changes? C-API doesn't understand (or see) parameter names. When I change it back manually there are no segfaults (and the dataset equivalent is still using the old variable name)

Reply:
I can have another looksee but the segfault happened after enabling this test: https://github.com/IQVIA-ML/LightGBM.jl/blob/master/test/basic_tests.jl#L89 where to get the `p_binary`  and `p_multi` we're using the `predict` which then uses `LGBM_BoosterPredictForMat` . This function had API changes as they also added `start_iteration` but this still didn't help to resolve the segfault. After renaming the parameter to what they have in their API, I didn't get the segfault any more and could actually see the julia stacktrace for the test itself. But yes  While the order of the parameters has to be respected I was surprised that the name change would make a difference (which you're right it shouldn't). In fact I didn't change the param name elsewhere...

Reply:
so yes it's not the `segfault` I was getting for missed API changes but I got the below error after I reversed this to be called `lgbm_data_type` and enabled the same test:
![image](https://user-images.githubusercontent.com/59925069/95068996-ff9e7780-06fd-11eb-8990-e4d3f2735ce0.png)

but after switching it back to `data_type` as in this PR, I actually got the error for the test itself before the full stacktrace and I didn't do any other changes:
![image](https://user-images.githubusercontent.com/59925069/95069098-2066cd00-06fe-11eb-953d-4ce48b7706b5.png)

Reply:
OK... something dodgy must have been happening locally on my machine. I cleaned up Manifest.toml and then run both cases and yes the name of the parameter doesn't make any difference. I brought back the `lgbm_data_type`